op-node/rollup/derive/batches.go

```diff
@@ -177,6 +178,10 @@ func checkSingularBatch(cfg *rollup.Config, log log.Logger, l1Blocks []eth.L1Blo
 			log.Warn("sequencers may not embed any SetCode transactions before Isthmus", "tx_index", i)
 			return BatchDrop
 		}
+		if !isL2Blob && txBytes[0] == types.BlobTxType {
```

Comment:
What about span batches, do we also need to validate the transactions within them?

Reply:
Then it seems a bit weird that the `checkSpanBatch` function also checks for DepositTxType [here](https://github.com/QuarkChain/optimism/blob/fdaa45e982da99318d915232b07e8e18ee1950ec/op-node/rollup/derive/batches.go#L367), given that checkSingularBatch already performs [the same check](https://github.com/QuarkChain/optimism/blob/fdaa45e982da99318d915232b07e8e18ee1950ec/op-node/rollup/derive/batches.go#L172)

Could you please share the code link that "span batches will be splitted into singular batches and go into the checkSingularBatch again"?

Reply:
It's in the `nextSingularBatchCandidate` function.